Move and split Connect to external resources (#2678)
astronomer/docs: astro/connect-aws.md

title: 'Connect Astro to AWS data sources'
id: connect-aws
description: Connect Astro to AWS resources.

> # toc_min_heading_level: 2
> # toc_max_heading_level: 2

Reviewer: ```suggestion
```

sidebar_custom_props: { icon: 'img/aws.png' }
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

> Based on the security considerations of your organization, your cloud resources might be accessible using public endpoints or from within a private network. Use the following topics to learn about different options available to access these resources from your Astro Deployment.

Reviewer: ```suggestion
Use this document to learn how you can connect an Astro cluster and its Deployments to your AWS resources.
```

Reviewer: All of these edits also apply to the other clouds (just need to swap out the cloud name)

## Connection options

> Publicly accessible endpoints allow you to quickly connect your Astro clusters or Deployments to AWS through only an Airflow connection. If you have restrictions based on IP address, you can add the external IPs of your Deployment or cluster to your cloud resource's allowlist. Whereas, to connect to private networks from Astro, you can choose from the available options provided by AWS.

Reviewer: ```suggestion
Publicly accessible endpoints allow you to quickly connect your Astro clusters or Deployments to AWS through an Airflow connection. If your cloud restricts IP addresses, you can add the external IPs of your Deployment or cluster to an AWS resource's allowlist. 

If you have stricter security requirements, you can [create a private connection](#create-a-private-connection-between-astro-and-aws) to AWS in a few different ways.

After you crate a connection from your Deployment to AWS, you might also have to individually authorize Deployments to access specific resources. See [Authorize your Deployment using workload identity](authorize-deployments-to-your-cloud.md#aws).
```

> ### Public endpoints

Reviewer: ```suggestion
### Access a public AWS endpoint
```

Reviewer: Unless it's a strategy or reference doc, we should always try to have action-oriented titles.

> To facilitate communication between your Astro cluster or Deployment with your external AWS resources, you can choose to allowlist them by retrieving their external IPs from the Cloud UI.

Reviewer: ```suggestion
To facilitate communication between your Astro cluster or Deployment and your cloud, you can allowlist the  external IPs for your cluster or Deployment on your cloud. If you have no other security restrictions, this means that any Deployment or cluster with an allowlisted external IP address can access your AWS resources trough a public endpoint.
```

#### Allowlist external IP addresses for a cluster

1. In the Cloud UI, click your Workspace name in the upper left corner, then click **Organization Settings**.
2. Click **Clusters**, then select a cluster.
3. In the Details page, copy the IP addresses listed under **External IPs**.
4. Add the IP addresses to the allowlist of any external services that you want your cluster to access.

> After you allowlist a cluster's IP address, all Deployments in that cluster are allowed to access your external resources.

Reviewer: ```suggestion
After you allowlist a cluster's IP address, all Deployments in that cluster are allowed to access your AWS resources.
```

#### Allowlist external IP addresses for a Deployment

To grant access to your external resources on per-Deployment basis, or if you are using a standard cluster, allowlist the IPs only for specific Deployments. For each Deployment that you want to allowlist:

1. In the Cloud UI, select a Deployment, then click Details.
2. Copy the IP addresses under External IPs.
3. Add the IP addresses to the allowlist of any external services that you want your Deployment to access.

When you use publicly accessible endpoints to connect to AWS, traffic moves directly between your Astro cluster and the AWS API endpoint. Data in this traffic never reaches the Astronomer-managed control plane.

> ### Private connections

Reviewer: ```suggestion
### Create a private connection between Astro and AWS
```

import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

To allow your Astro cluster to connect to your external data sources, you can either rely on public network or communicate using a private network. If you want to use a private network, there are multiple options to choose from. 

Use the following topics to learn about configuring various options available for connecting to your existing resources from Astro.

## Connection options

The option that you choose is determined by the security requirements of your company and your existing infrastructure.

<Tabs

>     defaultValue="Public endpoints"

Reviewer: I don't like public endpoints being in a tab alongside these other options. What if we had it as its own topic, and then kept all three private connection options under a separate "Private connections" tab? This would be more consistent with the structure we set up in the intro.

    ]}>

> <TabItem value="Public endpoints">

Reviewer: Per [this Slack message](https://astronomer.slack.com/archives/C015V2JFKT5/p1692366610627039), would this be the right place to mention the other domains to allowlist?

Publicly accessible endpoints allow you to quickly connect your Astro Deployments to AWS. In certain cases, you might need to whitelist your Astro cluster or Deployment to facilitate communication with your external AWS resources. You can choose to whitelist either your Astro cluster or just your Deployment by retrieveing the external IPs from your Cloud UI.

> ### Retrieve IP addresses for a cluster to whitelist

Reviewer: ```suggestion
### Allowlist external IP addresses for a cluster
```

1. In the Cloud UI, click the Astronomer logo in the top left corner to open your Organization.
2. Click **Clusters**, then select a cluster.
3. In the **Details** page, copy the IP addresses listed under **External IPs**.

A cluster's IP addresses are the same for all the Deployments running in that cluster. This is a one-time setup for each Astro cluster.

> ### Retrieve external IP addresses for a Deployment to whitelist

Reviewer: ```suggestion
### Allowlist external IP addresses for a Deployment
```

To allow to access to your external resource on per Deployment basis or if you are using a standard cluster, you need the external IPs of your Deployment.

1. In the Cloud UI, select a Deployment, then click **Details**.
2. Copy the IP addresses under **External IPs**.
3. (Optional) Add the IP addresses to the allowlist of any external services that need to interact with Astro.

> When you use publicly accessible endpoints to connect to AWS, traffic moves directly between your Astro cluster and the AWS API endpoint. Data in this traffic never reaches the Astronomer managed control plane.

Reviewer: ```suggestion
When you use publicly accessible endpoints to connect to AWS, traffic moves directly between your Astro cluster and the AWS API endpoint. Data in this traffic never reaches the Astronomer-managed control plane.
```
